lib/chess/auth/user.ex

```diff
@@ -0,0 +1,33 @@
+defmodule Chess.Auth.User do
+  @moduledoc false
+
+  use Ecto.Schema
+  import Ecto.Changeset
+  alias Comeonin.Argon2
+  alias Chess.Repo
+
+  schema "users" do
+    field :password, :string, virtual: true
+    field :password_hash, :string
+    field :username, :string
+
+    timestamps()
+  end
+
+  @doc false
+  def changeset(struct, params \\ %{}) do
+    struct
+    |> cast(params, [:username, :password])
+    |> validate_required([:username, :password])
+    |> unsafe_validate_unique(:username, Repo)
```

Comment:
What do you think of putting this unique constraint into the database? Then you can use [unique_constraint](https://hexdocs.pm/ecto/Ecto.Changeset.html#unique_constraint/3) and don't have to worry about race conditions.

Reply:
👍